chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb

```diff
@@ -71,6 +71,13 @@
 
   include_recipe "crowbar::prepare-upgrade-scripts"
 
+  bash "set pre-upgrade node attribute"
+    code <<-EOF
+      crm node attribute $(hostname) set pre-upgrade true
+    EOF
+    only_if { node["run_list_map"].key? "pacemaker-cluster-member" }
```

Comment:
A couple of things:

- generally I agree that we don't want to look at `run_list_map`; I know we did that in 5->6, but that's really digging in the implementation details in a way that we shouldn't
- since we don't deactivate proposals (I think), `node.role? "pacemaker-cluster-member"` should work
- if we want to be safe and not look at the roles, I would actually check if we have `node[:pacemaker][:config][:environment]`, although this would match remote pacemaker nodes, which might not be desired